docs/batch_request.md

```diff
@@ -28,3 +29,20 @@ batchItem2 := batch.AppendBatchItem(*reqInfo)
 
 batchItem1.DependsOnItem(batchItem2)
 ```
+
+## Adds BatchCollectionResponse
+
+`BatchRequestCollection` allows users to add more than 19 requests and send them as multiple `BatchRequest`'s. The send functionality of BatchRequestCollection splits the requests and sends them in serial.
+
+```go
+batchCollection := msgraphgocore.NewBatchRequestCollection(client.GetAdapter())
+
+meRequestItem, _ := batchCollection.AddBatchRequestStep(*meRequest)
+eventsRequestItem, _ := batchCollection.AddBatchRequestStep(*eventsRequest)
+
+batchResponse, _ := batchCollection.Send(context.Background(), client.GetAdapter())
+
+// print the first response
+user := GetBatchResponseById[User](resp, "1", CreateUserFromDiscriminatorValue) // returns a serialized response
```

Comment:
I'm guessing you meant this?
```suggestion
user := GetBatchResponseById[User](batchResponse, "1", CreateUserFromDiscriminatorValue) // returns a serialized response
```

but could we have this instead?

```suggestion
user := batchResponse.GetBatchResponseById[User]("1", CreateUserFromDiscriminatorValue) // returns a serialized response
```

Reply:
Yes the 1st one is correct.  The 2nd one will not be possible since it's a generic on a struct/interface method.

Reply:
ah right, the generics in Go are so limited...